kyaml/fn/framework/framework.go

```diff
@@ -56,6 +63,150 @@ type ResourceList struct {
 	Results Results `yaml:"results,omitempty" json:"results,omitempty"`
 }
 
+// ParseResourceList parses a ResourceList from the input byte array.
+func ParseResourceList(in []byte) (*ResourceList, error) {
```

Comment:
Similar to my comment below, I feel like this could at least be leveraging `kio.ByteReader` under the hood, and _maybe_ should take one as input (if having a constructor for that) depending on the circumstances under which we see this being useful. Could you clarify about that? This seems reasonable to have in general, but in practice I'd expect the framework to be taking care of parsing behind one of the higher-level entrypoints.